1-WebApp-OIDC/1-7-B2C-Embedded/README.md

```diff
@@ -0,0 +1,247 @@
+
+### Step 1: Clone or download this repository
+
+From your shell or command line:
+
+```powershell
+git clone https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2.git
+```
+
+> Given that the name of the sample is very long, and so are the names of the referenced NuGet packages, you might want to clone it in a folder close to the root of your hard drive, to avoid file size limitations on Windows.
+
+Navigate to the `"1-7-B2C-Embedded"` folder
+
+ ```Sh
+  cd "1-7-B2C-Embedded"
+  ```
+
+## Option 1 - Run the pre-configured sample
+
+1. Build the solution and run it.
+1. Open your web browser and make a request to the app. Accept the IIS Express SSL certificate if needed. Click on SignIn/Up button.
+1. Click on Sign-In
+
+## Option 2 (Recommended) - Configure the sample with your own B2C app
+
+### Step 2: Get your own Azure AD B2C tenant
+
+If you don't have an Azure AD B2C tenant yet, you'll need to create an Azure AD B2C tenant by following the [Tutorial: Create an Azure Active Directory B2C tenant](https://azure.microsoft.com/documentation/articles/active-directory-b2c-get-started).
+
+### Step 3: Enable custom domains for Azure Active Directory B2C
+
+To use Azure AD B2C Embedded sign-in, you need to first enable a custom domain with your Azure AD B2C account, and configure it with a custom domain which is on the same domain as your published application, i.e. **Same Origin**. This is required because Azure AD B2C should not allow you to load the sign-in experience in an iframe unless you enable CORS for the source domains loading the iframe using a custom policy. So first, enable a custom domain similar to **login.yourcustomdomain.com** if your application is published on something similar to **www.yourcustomdomain.com**. [Tutorial: Enable custom domains for Azure Active Directory B2C](https://docs.microsoft.com/en-us/azure/active-directory-b2c/custom-domain?pivots=b2c-custom-policy).
+
+>Note: This will effectively disallow you from using sign-in if the application is run from a development machine. To get around that, you may want to create a separate Azure AD Tenant for development, with a development version of the custom policy which has CORS enabled for **<https://localhost:XXXX>**. Also note that this may result in an undesirable user experience Because Azure AD B2C session cookies within an iframe are considered third-party cookies, certain browsers (for example Safari or Chrome in incognito mode) either block or clear these cookies. This should never be done in your production Azure AD B2C tenant.
```

Comment:
@helshabini this "Note" section needs a bit more hand holding. Do they really need a local development environment? If so, we should consider expanding the instructions here.

Reply:
@derisen The only reason I added this is to minimize the requirements for running the sample. Otherwise, users would need a complete deployment of the sample with real domains to try it out.

Reply:
@derisen  Sam suggesting using hosts file, which I think is a brilliant idea. I will test it out tomorrow and if things go fine I will commit further instructions for setting the sample locally.

Reply:
BTW for the hosts file, when you change domain.com to point to your own machine, be sure to change your local app server settings and your AAD app configuration's the reply address to be `https://localhost:443` or `http://localhost:80` depending on which protocol you're using.